File: frontend/src/types/governance.ts
export type AuditLogItem = {
  id: string;
  tenant_id: string | null;
  user_id: string | null;
  event_type: string;
  resource_type: string;
  resource_id: string | null;
  action: string;
  request_id: string | null;
  ip_address: string | null;
  payload: Record<string, unknown> | null;
  created_at: string;
};

export type ApprovalQueueItem = {
  id: string;
  run_id: string;
  tool_name: string;
  requested_by_user_id: string | null;
  approved_by_user_id: string | null;
  status: string;
  request_payload_json: Record<string, unknown>;
  decision_note: string | null;
  created_at: string;
  decided_at: string | null;
};

File: backend/app/schemas/governance.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class AuditLogRead(BaseModel):
    id: str
    tenant_id: str | None = None
    user_id: str | None = None
    event_type: str
    resource_type: str
    resource_id: str | None = None
    action: str
    request_id: str | None = None
    ip_address: str | None = None
    payload: dict[str, Any] | None = None
    created_at: datetime


class AuditLogQuery(BaseModel):
    tenant_id: str | None = None
    user_id: str | None = None
    event_type: str | None = None
    tool: str | None = None
    start_at: datetime | None = None
    end_at: datetime | None = None
    limit: int = Field(default=100, ge=1, le=500)
    offset: int = Field(default=0, ge=0)
